### PR TITLE
Read cubefile xs, ys, zs correctly

### PR DESCRIPTION
**Pull request by tillhanke**
* Read cubefile nx and use that to calculate xs (y and z as well)

* Fix issue #1531

### CODE REVIEW

**Comment by tillhanke** [comment]
I am contributing via pull request basically for the first time ever. I hope I did this the right way.

Feedback would be very welcome ;)

File: pyscf/tools/cubegen.py [hunk 1]
```diff
@@ -326,12 +326,23 @@ def read(self, cube_file):
...
+            def parse_nx(data, x):
+                from pyscf.pbc.gto import Cell
                 d = data.split()
-                return int(d[0]), numpy.array([float(x) for x in d[1:]])
-            self.nx, self.xs = parse_nx(f.readline())
-            self.ny, self.ys = parse_nx(f.readline())
-            self.nz, self.zs = parse_nx(f.readline())
+                nx = int(d[0])
+                x_vec = numpy.array([float(x) for x in d[1:]])
+                assert (numpy.delete(x_vec, x) == 0).all()
```
Comment (fishjojo) [inline pyscf/tools/cubegen.py:334]: This is not true if the box is not rectangular.

Comment (tillhanke) [reply pyscf/tools/cubegen.py:334]: But isn't the `Cube` object only usable with rectangular boxes?
I know, that cubefiles can theoretically use non rectangular boxes. But (as far as I know) other than throwing an error at this point the `Cube` object can't handle that.

Or am I missing something?

Comment (tillhanke) [reply pyscf/tools/cubegen.py:334]: As a reference I have found 
https://github.com/pyscf/pyscf/blob/3c8c7875fdc0921ee4aec0338ca1ae3413971288/pyscf/tools/cubegen.py#L247-L250

Comment (fishjojo) [reply pyscf/tools/cubegen.py:334]: For periodic systems, the box is the unit cell.
https://github.com/pyscf/pyscf/blob/3c8c7875fdc0921ee4aec0338ca1ae3413971288/pyscf/tools/cubegen.py#L241-L242

Comment (tillhanke) [reply pyscf/tools/cubegen.py]: thanks for the input! I think I fixed that now.

File: pyscf/tools/cubegen.py [hunk 2]
```diff
@@ -326,12 +326,23 @@ def read(self, cube_file):
             data = f.readline().split()
             natm = int(data[0])
             self.boxorig = numpy.array([float(x) for x in data[1:]])
-            def parse_nx(data):
+            def parse_nx(data, x):
```
Comment (sunqm) [inline pyscf/tools/cubegen.py:329]: the argument `x` is not needed

**Review by sunqm — APPROVED** [review]

**Pull request merged**